### DISCUSSION TITLE
Klipper version

### DISCUSSION THREAD

**Discussion by mindkill666** [Guilouz/Creality-Helper-Script-Wiki#475] [General]
Hi, i was just wondering if klipper is updated automaticly to the newest version.
I'm thinking about using a carthographer probe, but when i update klipper manually from https://github.com/K1-Klipper/installer_script_k1_and_max, i get errors.

On their site they say that klipper needs to be latest version, so i wanted to know if klipper is the latest version with your script.

**Comment by justacec** [comment]
I am having the same question here.

I followed this help script guide, but then realized that it did not bring Klipper to current.  When I ran the scripts at the OP's provided URL, I got issues with a virtual pin.  

Is the updated forked Klipper not compatible with these provided helper scripts or is there a specific order of operations what must be completed, or was the updated Klipper broken itself?

**Comment by Guilouz** [comment]
Virtual pins is compatible with all version of Klipper, just to be sure to add an empty [virtual_pins] section to your printer.cfg to enable the virtual_pins

**Comment by halTWO3** [comment]
Same issue here. Even though virtual_pins already defined in config, still getting unknown name error.

**Comment by justacec** [comment]
Ok, So I got this error by doing the following procedure:

1: Restore to factory default
2: Upgrade firmware to latest from the K1 Max touch screen
3: Root the machine through the Creality touch screen
4: Install Fluidd and Moonraker along with adding several of the different helper scripts from https://guilouz.github.io/Creality-Helper-Script-Wiki/
5: Reboot a few times
6: Attempt to upgrade Klipper to the more current version by following the guide at: https://github.com/K1-Klipper/installer_script_k1_and_max

**Reply by Guilouz** [reply]
Virtual pins is only installed when installing kamp. If you have only installed moonraker, fluidd and updated klipper you can’t have this error

**Reply by justacec** [reply]
Humm.  Yea, I had installed KAMP.  That was part of the "different helper scripts".  It seemed that after I tried to update Klipper, it did not know about the [virtual_pins] directive at all.  I am also a bit concerned that the K1-Klipper repository seems to be archived now as of approximately 3 weeks ago.  Does this mean that it is no longer being maintained?  It is seeming more and more like I will be unable to use your great helper script in addition to the updated Klipper and just use the most recent Creality firmware (rooted with your script of course)

**Reply by halTWO3** [reply]
Edit: You beat me to it. I removed gcode shell then the install went thru.

**Reply by justacec** [reply]
I was able to finally get things to work.  Still a few bugs to work out but it is in the Klipper config.  Steps to do:

1: Reset to factory default
2: Ensure you have latest firmware
3: Root the machine
4: Install Fluidd and Moonraker and the Entware package manager only  from https://guilouz.github.io/Creality-Helper-Script-Wiki/
5: Upgrade Klipper to the more current version by following the guide at: https://github.com/K1-Klipper/installer_script_k1_and_max
6: Go back in and install additional helper scripts with the Guilouz helper script installer

This will get you a Klipper from ~Apr 8.  But I am still confused as to why the K1-Klipper repo is archived and I am a bit concerned that means that he will no longer keep that up to date.  I am not sure what the motivations for that would be.  I am hopeful that there is some way that Klipper can be kept up to date.

**Reply by Guilouz** [reply]
This repo will not be updated, new PR cause some issue on K1 with prtouch.

**Reply by justacec** [reply]
Humm.  That is a bummer.  I guess I will forgo trying to do the updated Klipper and just stick with the stock FW with your helper script to help open it up.  I will keep my fingers crossed that there will be a future effort to bring in the updated Klipper.  I would hate to be stuck with the 2022 Klipper when there has been so much that has gone on with it.

**Reply by dfbpurcell** [reply]
> This repo will not be updated, new PR cause some issue on K1 with prtouch.

What issue? I am rooting my machine because I have been suffering from a super inconsistent z offset that does not respect the bed mesh or my saved settings, destroyed 7 bed plates and failed every print in the past 6 months from the day I bought it new
